Bump normaliz to 3.10.5

d-torrance opened the pull request:
This was just released.  We also make `--enable-download` the default behavior in the autotools build.

Draft for now to test the GitHub builds.

d-torrance committed:
Make --enable-download the default configure option

d-torrance committed:
Bump normaliz to 3.10.5

mahrud commented on M2/Macaulay2/packages/Normaliz.m2:
What's changing here?

d-torrance replied:
One of the outputs changed from "multiplicity" to "multiplicity num"

d-torrance committed:
Wrangle normaliz 3.10.5 output to match older versions

"multiplicity num" is the new name for "multiplicity" and "hilbert
series cyclo denom" and "hilbert series cyclo num" were added, but we
rename/remove them for backwards compatibility.

Also update loop to use "for ... list" instead of repeatedly calling
append.

d-torrance commented:
Yay, builds passed!

https://github.com/Macaulay2/M2/actions/runs/15456617018?pr=3871
